add: with_lookup pitfalls (#61)
qdrant/skills: skills/qdrant-search-quality/search-strategies/hybrid-search/search-types/SKILL.md

When a representation's signal is mostly lexical — keyword-driven titles, codes, tags, or other short fields — prefer a sparse named vector (e.g. BM25) over an additional dense embedding. Server-side BM25 in Qdrant avoids the inference cost of another dense model and stores far less per point. Skip this when the field carries paraphrase or conceptual signal that exact-term matching would miss.

- End-to-end worked example fusing title, abstract, chunk, and sparse-title named vectors with RRF and document-level grouping in one Query API call: [Multi-Representation Search tutorial](https://search.qdrant.tech/md/documentation/tutorials-search-engineering/multi-representation-search/)

> - If you have groups and subgroups of representations (document -> chunk, image -> patch), you could use [searching in groups](https://search.qdrant.tech/md/documentation/search/search/?s=search-groups). Index the grouping payload field (e.g. `document_id`) with a schema matching its value type — `keyword` for string-valued IDs, `integer` for numeric IDs — before grouping. For `with_lookup`, the `group_by` value must also be a valid Qdrant point ID (unsigned integer or UUID string) so it can match a point in the lookup collection. Arbitrary string IDs like DOIs or slugs are fine as payload but cannot be point IDs — map them to a stable integer or UUID join key stored in both collections before using `with_lookup`.

Reviewer: What do you think about splitting this into several sub-bullets?
```suggestion
- If you have groups and subgroups of representations (document -> chunk, image -> patch), you could use [searching in groups](https://search.qdrant.tech/md/documentation/search/search/?s=search-groups): 
  - Index the grouping payload field (e.g. `document_id`) with a schema matching its value type — `keyword` for string-valued IDs, `integer` for numeric IDs — before grouping.
  - For `with_lookup`, ensure the `group_by` value is a valid Qdrant point ID (unsigned integer or UUID string) and matches the point in the lookup collection.
  - Arbitrary string IDs like DOIs or slugs cannot be point IDs — map them to a stable integer or UUID join key stored in both collections before using `with_lookup`.
```

Author: fixed in [42f0ce9](https://github.com/qdrant/skills/pull/61/commits/42f0ce9a0b71b1eb77a5df58f89dabe3b7c3a300)

- When grouping chunk-level points back to documents, each prefetch only contributes the candidates it returned — so size per-prefetch `limit` well above the final document `limit` (rule of thumb: `prefetch_limit ≥ final_limit × expected_chunks_per_document`), otherwise a few documents with many chunks saturate the candidate pool and relevant documents drop silently. Validate grouped recall on a labeled sample.

### When to Split into a Sidecar Collection with Lookup in Groups

> Duplicated document-level data across chunk points scales as `documents × chunks_per_document × shared_bytes_per_point` (e.g. 20k docs × 24 chunks × ~3 KB ≈ 1.4 GB duplicated vs. ~60 MB once split). Whether to split depends on what the shared data is used for, not just its size. Classify before recommending [Lookup in Groups](https://search.qdrant.tech/md/documentation/search/search/?s=lookup-in-groups):

Reviewer: I think this sentence misses a subject.

```suggestion
Duplicated document-level data across chunk points scales as `documents × chunks_per_document × shared_bytes_per_point` (e.g. 20k docs × 24 chunks × ~3 KB ≈ 1.4 GB duplicated vs ~60 MB once split). Whether to split depends on what the shared data is used for, not just its size. Classify the shared data before recommending [Lookup in Groups](https://search.qdrant.tech/md/documentation/search/search/?s=lookup-in-groups):
```

Author: fixed in [42f0ce9](https://github.com/qdrant/skills/pull/61/commits/42f0ce9a0b71b1eb77a5df58f89dabe3b7c3a300)

- **Shared data includes vectors (title, abstract, summary) that should contribute candidates or scores during retrieval** → keep denormalized on chunk points. `with_lookup` joins after grouping, so sidecar vectors never enter prefetch or fusion. Moving them silently degrades recall and ranking while queries still return results.
- **Shared data includes fields used in server-side filters or scoring** (tenant ID, ACL/visibility flags, status, publication date for recency boosts, anything referenced in Qdrant `filter` conditions or `FormulaQuery` expressions) → keep denormalized on chunk points. Filters and score formulas run during candidate generation, before grouping; sidecar values are invisible to them and the chunks remain unfiltered.
- **Mixed case** (per-document vector or filter field needed for retrieval + heavy per-document payload only for display) → keep the retrieval-relevant vector or filter field denormalized on chunks and move only the heavy display payload to the sidecar.

> Before activating Lookup in Groups, verify the preconditions — failures here return empty `lookup` fields with no error:

Reviewer: I'm not sure what `failures` refers to here. Activating or verifying? I suggest something along these lines:
```suggestion
Before activating Lookup in Groups, verify the preconditions — unmet conditions return empty `lookup` fields silently with no error:
```

Author: fixed in [42f0ce9](https://github.com/qdrant/skills/pull/61/commits/42f0ce9a0b71b1eb77a5df58f89dabe3b7c3a300)

When a representation's signal is mostly lexical — keyword-driven titles, codes, tags, or other short fields — prefer a sparse named vector (e.g. BM25) over an additional dense embedding. Server-side BM25 in Qdrant avoids the inference cost of another dense model and stores far less per point. Skip this when the field carries paraphrase or conceptual signal that exact-term matching would miss.

- End-to-end worked example fusing title, abstract, chunk, and sparse-title named vectors with RRF and document-level grouping in one Query API call: [Multi-Representation Search tutorial](https://search.qdrant.tech/md/documentation/tutorials-search-engineering/multi-representation-search/)

> - If you have groups and subgroups of representations (document -> chunk, image -> patch), you could use [searching in groups](https://search.qdrant.tech/md/documentation/search/search/?s=search-groups):

Reviewer 2: maybe `https://skills.qdrant.tech/md/documentation/search/search?s=grouping-api` better?

Author: Done

  - Index the grouping payload field (e.g. `document_id`) with a schema matching its value type — `keyword` for string-valued IDs, `integer` for numeric IDs — before grouping.
  - For `with_lookup`, ensure the `group_by` value is a valid Qdrant point ID (unsigned integer or UUID string) and matches the point in the lookup collection.
  - Arbitrary string IDs like DOIs or slugs cannot be point IDs — map them to a stable integer or UUID join key stored in both collections before using `with_lookup`.
- When grouping chunk-level points back to documents, each prefetch only contributes the candidates it returned — so size per-prefetch `limit` well above the final document `limit` (rule of thumb: `prefetch_limit ≥ final_limit × expected_chunks_per_document`), otherwise a few documents with many chunks saturate the candidate pool and relevant documents drop silently. Validate grouped recall on a labeled sample.

> ### When to Split into a Sidecar Collection with Lookup in Groups

Reviewer 2: it feels unrelated to hybrid seach. Maybe a separate skill?

Reviewer 2: Also keeping skills smaller seems more beneficial

Author: Pulled grouping + sidecar/with_lookup into its own leaf, search-strategies/document-grouping/, as a sibling to hybrid-search/.

Eval id 7 now targets the new leaf.

> Duplicated document-level data across chunk points scales as `documents × chunks_per_document × shared_bytes_per_point` (e.g. 20k docs × 24 chunks × ~3 KB ≈ 1.4 GB duplicated vs ~60 MB once split). Whether to split depends on what the shared data is used for, not just its size. Classify the shared data before recommending [Lookup in Groups](https://search.qdrant.tech/md/documentation/search/search/?s=lookup-in-groups):

Reviewer 2: should not use absolute link to `search.` subdomain. This triggers agent's built-in protectins about using unknown domains

Author: Done, though flagging that this conflicts with our guidelines in CONTRIBUTING.md and AGENTS.md, both currently mandate search.qdrant.tech/md/documentation/.
